subscription-fee/src/daily_operations.rs

```diff
@@ -0,0 +1,233 @@
+use auto_farm::common::{address_to_id_mapper::AddressId, unique_payments::UniquePayments};
+use multiversx_sc_modules::ongoing_operation::{CONTINUE_OP, STOP_OP};
+
+use crate::service::SubscriptionType;
+
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+pub type Epoch = u64;
+
+pub const DAILY_EPOCHS: Epoch = 1;
+pub const WEEKLY_EPOCHS: Epoch = 7;
+pub const MONTHLY_EPOCHS: Epoch = 30;
+
+pub const FIRST_INDEX: usize = 1;
+pub const GAS_TO_SAVE_PROGRESS: u64 = 100_000;
+
+#[derive(TypeAbi, TopEncode, TopDecode, Default)]
+pub struct OperationProgress {
+    pub service_id: AddressId,
+    pub service_index: usize,
+    pub current_index: usize,
+}
+
+#[multiversx_sc::module]
+pub trait DailyOperationsModule:
+    crate::fees::FeesModule
+    + crate::user_tokens::UserTokensModule
+    + crate::service::ServiceModule
+    + crate::pair_actions::PairActionsModule
+    + crate::low_level_actions::LowLevelActionsModule
+    + energy_query::EnergyQueryModule
+    + multiversx_sc_modules::ongoing_operation::OngoingOperationModule
+{
+    #[endpoint(performService)]
+    fn perform_service(&self, service_index: usize) -> OperationCompletionStatus {
```

Comment:
this has to be in another contract.

Reply:
plus we need the premiumaccountSC to interact with subscription contract, does that contract to make the registering for service, adding 2 service types.